data-prepper-core/src/test/java/org/opensearch/dataprepper/pipeline/router/DataFlowComponentRouterTest.java

```diff
@@ -51,14 +54,30 @@ private static class TestComponent {
     @BeforeEach
     void setUp() {
         recordsIn = Collections.emptyList();
-        routerCopyRecordStrategy = mock(RouterCopyRecordStrategy.class);
-        when(dataFlowComponent.getComponent()).thenReturn(testComponent);
+        lenient().when(dataFlowComponent.getComponent()).thenReturn(testComponent);
+        getRecordStrategy = mock(RouterGetRecordStrategy.class);
+        lenient().when(getRecordStrategy.getRecord(any())).thenAnswer(a -> a.getArgument(0));
+        lenient().when(getRecordStrategy.getAllRecords(anyCollection())).thenAnswer(a -> a.getArgument(0));
     }
 
     private DataFlowComponentRouter createObjectUnderTest() {
         return new DataFlowComponentRouter();
     }
 
+    @Test
+    void route_with_null_strategy_throws() {
+        getRecordStrategy = null;
+        assertThrows(NullPointerException.class, () -> createObjectUnderTest().route(recordsIn, dataFlowComponent, null, getRecordStrategy, componentRecordsConsumer));
```

Comment:
We should fail quickly. So you should check for a null pointer in the constructor and change this test to:

```
assertThrows(NullPointerException.class, () -> createObjectUnderTest());
```

In the current code and test, we will not know if there is an issue until after the pipeline is running without that null check.